core/src/test/kotlin/api/AdminRouteIntegrationTest.kt

```diff
@@ -42,4 +61,74 @@ class AdminRouteIntegrationTest : AbstractIntegrationTest({
             }
         }
     }
+
+    "GET /admin/users" should {
+        "return a list of users" {
+            integrationTestApplication {
+                val response = superuserClient.get("/api/v1/admin/users")
+
+                response shouldHaveStatus HttpStatusCode.OK
+                val users = Json.decodeFromString<Set<User>>(response.bodyAsText())
+                users.shouldNotBeNull()
+                users shouldContain User(
+                    username = SUPERUSER.username.value,
+                    firstName = SUPERUSER.firstName,
+                    lastName = SUPERUSER.lastName,
+                    email = SUPERUSER.email
+                )
+            }
+        }
+    }
+
+    "POST /admin/users" should {
+        "create a new user" {
+            integrationTestApplication {
+                val user = CreateUser(testUsername, testPassword, testTemporary)
+
+                val response = superuserClient.post("/api/v1/admin/users") {
+                    setBody(user)
+                }
+
+                response shouldHaveStatus HttpStatusCode.Created
+            }
+        }
+        "respond with an internal error if the user already exists" {
+            integrationTestApplication {
+                val user = CreateUser(testUsername, testPassword, testTemporary)
+
+                superuserClient.post("/api/v1/admin/users") {
+                    setBody(user)
+                }
+                val response = superuserClient.post("/api/v1/admin/users") {
+                    setBody(user)
+                }
+
+                response shouldHaveStatus HttpStatusCode.InternalServerError
+            }
+        }
+    }
+
+    "DELETE /admin/users" should {
+        "delete a user" {
+            integrationTestApplication {
+                val response = superuserClient.delete("/api/v1/admin/users") {
+                    parameter("username", TEST_USER.username.value)
+                }
+
+                response shouldHaveStatus HttpStatusCode.NoContent
+            }
+        }
+        "respond with an internal error if the user doesn't exist" {
+            integrationTestApplication {
+                superuserClient.delete("/api/v1/admin/users") {
+                    parameter("username", TEST_USER.username.value)
+                }
+                val response = superuserClient.delete("/api/v1/admin/users") {
+                    parameter("username", TEST_USER.username.value)
+                }
+
+                response shouldHaveStatus HttpStatusCode.InternalServerError
+            }
+        }
+    }
```

Comment:
There should also be a test for each endpoint to verify that the superuser role is required (like for `GET /admin/sync-roles`).